File: DataStructure/20-Valid-Parentheses.cpp
#include <stack>
class Solution {
public:
    stack<int> Stack;
    bool isValid(string s) {
        for (int i = 0; s[i] != '\\0'; i++)
        {
            if(Stack.empty())
            {
                Stack.push(s[i]);
            }else if((Stack.top() == '(' && s[i] == ')') || (Stack.top() == '{' && s[i] == '}') || (Stack.top() == '[' && s[i] == ']'))
            {
                Stack.pop();
            }else{
                Stack.push(s[i]);
            }
        }
        return Stack.empty();
    }
};

File: DataStructure/20-Valid-Parentheses.py
class Solution(object):
    def isValid(self, s):
        \\\
        :type s: str
        :rtype: bool
        \\\
        stack = []
        for i in s:
            if i in \({[\:
                stack.append(i)
            else:
                if not stack:
                    return False
                if (stack[-1] == \(\ and i == \)\) or (stack[-1] == \{\ and i == \}\) or (stack[-1] == \[\ and i == \]\):
                    stack.pop()
                else:
                    return False

        return not stack       